### PR TITLE
Improve RSS feed

### PR DESCRIPTION
**Pull request by veger**
* Add self link
* Add category/categories if available
* Add image(s) if page_images plugin is available

The resulting feed should be validatable by https://validator.w3.org/feed/

### CODE REVIEW

**Comment by veger** [comment]
Additionally the following could be added to the channel element
```ruby
    xml.image do
      xml.url url_to_website_logo
      xml.title Refinery::Core.site_name
      xml.link refinery.blog_root_url
    end
```
but AFAIK there is not default location where a logo/image is available..?

**Comment by veger** [comment]
Build fails with "An error occurred while installing nokogiri (1.7.0.1)" for Ruby 2.0.0 targets

File: app/views/refinery/blog/posts/index.rss.builder [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
     @posts.each do |post|
       xml.item do
         xml.title post.title
-        xml.description post.body
+        xml.description blog_post_teaser(post)
+        xml.category post.categories.map {|x| x.title }.join('/') if  post.categories.length > 0
         xml.pubDate post.published_at.to_s(:rfc822)
         xml.link refinery.blog_post_url(post)
+        xml.guid refinery.blog_post_url(post)
+        post.images.each do |image|
```
Comment (parndt) [inline app/views/refinery/blog/posts/index.rss.builder]: Blog posts don't have an images association?

Comment (veger) [reply app/views/refinery/blog/posts/index.rss.builder]: It is added by refinerycms-page-images...
I guess we can either
* remove image support from the RSS
* add some check if `images` is available

Any preference on this matter?

Comment (parndt) [reply app/views/refinery/blog/posts/index.rss.builder]: I'm happy to just add some check if it's available as to me it seems useful and a somewhat expected use-case

File: app/views/refinery/blog/posts/index.rss.builder [hunk 2]
```diff
@@ -14,8 +14,10 @@ xml.rss version: "2.0", 'xmlns:atom': 'http://www.w3.org/2005/Atom', 'xmlns:medi
         xml.pubDate post.published_at.to_s(:rfc822)
         xml.link refinery.blog_post_url(post)
         xml.guid refinery.blog_post_url(post)
-        post.images.each do |image|
-          xml.media :content, url: refinery.root_url + image.url, medium: 'image'
+        if Refinery::Plugins.registered['page_images']
```
Comment (bricesanchez) [inline app/views/refinery/blog/posts/index.rss.builder]: `post.images` could come from another extension, custom code, i think we should only do:

```ruby
post.images.each do |image|
  xml.media :content, url: refinery.root_url + image.url, medium: 'image'
end if post.images.present?
```

Comment (veger) [reply app/views/refinery/blog/posts/index.rss.builder]: I thought about this, but then we do not know (for certain) that images is an array and its items have the url attribute... (or we need to more checks)

Comment (parndt) [reply app/views/refinery/blog/posts/index.rss.builder]: hmm 🤔 

Is there any way that the `refinerycms-page-images` extension can add this in without overriding the entire file?

Comment (veger) [reply app/views/refinery/blog/posts/index.rss.builder]: only thing I can/could think of, is to create helper function(s) to register custom parts of the feed (e.g. image(s)) in the refinery core (so move feed feature to core). And let the different plugin(s) use those functions to build/customize the feed according to their (combined) needs...

But this is a big, complex overkill just to have a feed (IMO)

Comment (parndt) [reply app/views/refinery/blog/posts/index.rss.builder]: I agree, that is quite a lot more work! I'm happy to just assume the default case of:

```ruby
post.images.each do |image|
  xml.media :content, url: refinery.root_url + image.url, medium: 'image'
end if post.respond_to?(:images)
```

Or whatever it takes to check whether we actually have an images association.

In terms of an image having a URL, I'd say that we can just assume that's the contract for a blog image? 

What do you think?

Comment (veger) [reply app/views/refinery/blog/posts/index.rss.builder]: Agree, let's not make it overly complex...!

I'll update the PR

**Comment by bricesanchez** [comment]
Thank you @veger !

**Pull request merged**